lib/iris/tests/idiff.py

```diff
@@ -44,49 +79,114 @@ def diff_viewer(expected_fname, result_fname, diff_fname):
     ax = plt.subplot(223, sharex=ax, sharey=ax)
     ax.imshow(mimg.imread(diff_fname))
 
+    # Determine the new image hash.png name.
+    with open(result_fname, 'rb') as fi:
+        sha1 = hashlib.sha1(fi.read())
+    hash_fname = os.path.join(dname, _HASH_DIR, sha1.hexdigest()+'.png')
+    uri = os.path.join(_PREFIX_URI, os.path.basename(hash_fname))
+
     def accept(event):
-        # removes the expected result, and move the most recent result in
-        print('ACCEPTED NEW FILE: %s' % (os.path.basename(expected_fname), ))
-        os.remove(expected_fname)
-        shutil.copy2(result_fname, expected_fname)
+        if uri not in repo[key]:
+            # Ensure to maintain strict time order where the first uri
+            # associated with the repo key is the oldest, and the last
+            # uri is the youngest
+            repo[key].append(uri)
+            with open(os.path.join(dname, _POSTFIX_JSON), 'wb') as fo:
+                json.dump(repo, fo, indent=4, sort_keys=True)
+            shutil.copy2(result_fname, hash_fname)
+            msg = 'ACCEPTED:  {} -> {}'
+            print(msg.format(os.path.basename(result_fname),
+                             os.path.basename(hash_fname)))
+        else:
+            msg = 'DUPLICATE: {} -> {} (ignored)'
+            print(msg.format(os.path.basename(result_fname),
+                             os.path.basename(hash_fname)))
+        os.remove(result_fname)
         os.remove(diff_fname)
         plt.close()
 
     def reject(event):
-        print('REJECTED: %s' % (os.path.basename(expected_fname), ))
+        if uri not in repo[key]:
+            print('REJECTED:  {}'.format(os.path.basename(result_fname)))
+        else:
+            msg = 'DUPLICATE: {} -> {} (ignored)'
+            print(msg.format(os.path.basename(result_fname),
+                             os.path.basename(hash_fname)))
+            os.remove(result_fname)
+        os.remove(diff_fname)
         plt.close()
 
-    ax_accept = plt.axes([0.7, 0.05, 0.1, 0.075])
-    ax_reject = plt.axes([0.81, 0.05, 0.1, 0.075])
-    bnext = mwidget.Button(ax_accept, 'Accept change')
-    bnext.on_clicked(accept)
-    bprev = mwidget.Button(ax_reject, 'Reject')
-    bprev.on_clicked(reject)
+    def skip(event):
+        print('SKIPPED:   {}'.format(os.path.basename(result_fname)))
+        os.remove(diff_fname)
```

Comment:
I think this should leave the result in-place, so we can deal with it later by re-running
(see general behaviour discussion)